src/v0/destinations/criteo_audience/transform.js

```diff
@@ -37,19 +37,19 @@ const responseBuilder = (message, destination, accessToken) => {
   const { Config } = destination;
   const { audienceId } = Config;
   const { listData } = message.properties;
-  operation.forEach(op => {
+
+  let finalAudienceId = audienceId;
+  const mappedToDestination = get(message, MappedToDestinationKey);
+  if (!finalAudienceId && mappedToDestination) {
+    const { objectType } = getDestinationExternalIDInfoForRetl(message, 'CRITEO_AUDIENCE');
```

Comment:
As we didn't make the `audienceId` mandatory in the configuration flow, is this additional logic being required ?

Reply:
here we are checking if audienceId is present in the config itself or in the context. 
As we already have vdm enabled and now we have audienceId in the dest config such existing connection will not have audienceid in the context. so this is for backward compatibilty. (though we have not announced criteo aud yet so very less chance of such cases)